### PR TITLE
Add Calendar functionality

### PR DESCRIPTION
**Pull request by shawnlsj97**
Able to display days and months correctly. Current navigation is via clicking of left and right arrows. Navigation by command will be introduced in the following days, as well as indicating training and performance records.

### CODE REVIEW

File: src/main/java/seedu/address/ui/CalendarPanel.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        neededDays = reverse(neededDays);
+        return neededDays;
+    }
+
+    /**
+     * Reverses a String array.
+     * @param s String array to be reversed
+     * @return Reversed String array.
+     */
+    private String[] reverse(String[] s) {
```
Comment (hellodommy) [inline src/main/java/seedu/address/ui/CalendarPanel.java:206]: I assume this is populating the days of the previous month from the back? Good work!

File: src/main/java/seedu/address/ui/CalendarPanel.java [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+    /**
+     * Combines 3 string arrays together, with array {@code a}'s elements coming in front, array
+     * {@code b}'s elements coming in the middle and array {@code c}'s elements coming at the back.
+     * @param a String array whose elements are to be in front
+     * @param b String array whose elements are to be in the middle
+     * @param c String array whose elements are to be behind
+     * @return String[] Combined String array.
+     */
+    private String[] combine(String[] a, String[] b, String[] c) {
```
Comment (hellodommy) [inline src/main/java/seedu/address/ui/CalendarPanel.java:266]: Perhaps can consider giving `a`, `b` and `c` more meaningful names like `previous month`, `current month` and `next month`.

File: src/main/java/seedu/address/ui/CalendarPanel.java [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+            for (int j = 0; j < 7; j++) {
+                int column = j;
+                HBox h = new HBox();
+                Label l = new Label();
+                l.setText(days[counter]);
+                l.setPadding(new Insets(5, 0, 0, 10));
+                l.setFont(new Font("System", 11));
+                l.setAlignment(Pos.TOP_LEFT);
+                if (beforeCount > 0) {
+                    l.setTextFill(Paint.valueOf("#999999"));
```
Comment (hellodommy) [inline src/main/java/seedu/address/ui/CalendarPanel.java:296]: Great attention to detail in making the colour of the days of the previous month lighter.

**Review by hellodommy — COMMENTED** [review]
Overall, great job in populating the calendar. This will be pivotal in our following feature implementations. 👏🏻

File: src/main/java/seedu/address/ui/CalendarPanel.java (hunk 3 above)
Comment (shawnlsj97) [reply src/main/java/seedu/address/ui/CalendarPanel.java:296]: thanks!

File: src/main/java/seedu/address/ui/CalendarPanel.java (hunk 1 above)
Comment (shawnlsj97) [reply src/main/java/seedu/address/ui/CalendarPanel.java:206]: yup! it's an innovative approach!

File: src/main/java/seedu/address/ui/CalendarPanel.java (hunk 2 above)
Comment (shawnlsj97) [reply src/main/java/seedu/address/ui/CalendarPanel.java:266]: alright will make the changes!

**Comment by hellodommy** [comment]
Not possible to have test classes for UI-related classes, hence the drop in Coveralls.

**Comment by hellodommy** [comment]
Not possible to have test classes for UI-related classes, hence the drop in Coveralls.

**Pull request merged**